src/zkevm_specs/evm_circuit/execution/memory.py

```diff
@@ -43,3 +65,58 @@ def memory(instruction: Instruction):
         memory_word_size=Transition.to(next_memory_size),
         dynamic_gas_cost=memory_expansion_gas_cost,
     )
+
+
+def check_mask(offset, M):
+    # Compute 2**offset by squaring-and-multiplying.
+    offset_value = 1
+    for i in reversed(range(5)):
+        bit = (offset >> i) & 1
+        offset_value = offset_value * offset_value * (1 + bit)
+
+    # Interpret the mask as a binary number.
+    mask_value = 0
+    for (i, m) in enumerate(M):
```

Comment:
need to check `M` begin with consecutive 1 explicitly ?

Reply:
The assert with offset_value below is supposed do that.